src/main.rs

```diff
@@ -165,6 +206,32 @@ mod tests {
     use sandboxfs::Mapping;
     use super::*;
 
+    /// Checks that an error, once formatted for printing, contains the given substring.
+    fn err_contains<F: failure::Fail>(substr: &str, err: F) {
+        let formatted = format!("{}", err);
+        assert!(formatted.contains(substr),
+            "bad error message '{}'; does not contain '{}'", formatted, substr);
+    }
+
+    #[test]
+    fn test_parse_duration_ok() {
+        assert_eq!(Timespec { sec: 1234, nsec: 0 }, parse_duration("1234s").unwrap());
+    }
+
+    #[test]
+    fn test_parse_duration_bad_unit() {
+        err_contains("must end in 's'", parse_duration("1234").unwrap_err());
+        err_contains("invalid digit", parse_duration("1234ms").unwrap_err());
+        err_contains("invalid digit", parse_duration("1234ss").unwrap_err());
```

Comment:
This isn't really the best error.

Reply:
Fixed this instance. But following your new proposal, some other errors have become not great either. Don't care much though.